### ISSUE TITLE
Notepad++ search results reset

### ISSUE THREAD

**Issue by mididoc** [notepad-plus-plus/notepad-plus-plus#14107]
Hi,
i search something in notepad++ and get result in multiple files.
i may select all and all these files open, where the search is.
now i search for another thing and get result in multiple files.
if i click now on 'open all' notepad++ opens the files of current search but also the files of recent search.
that mean, notepad++ does not reset the files result.
if i search something, i want always only the cresponding files and not the files of recent searches.
this is reproducable.
is there an option to reset?
thanks for assistance.
cheers mike

**Comment by alankilborn** [comment]
>is there an option to reset?

Right-click in the _Search results_ window, then choose _Purge for every search_ from the resulting popup menu:

![image](https://github.com/notepad-plus-plus/notepad-plus-plus/assets/77065706/a8f0facf-fe2a-4ed0-b9f8-9f33133bebcc)

**Comment by mididoc** [comment]
alankilborn,
thanks lot, this is really great.
didn't know this option.
cheers mike

**Comment by vijaysitaram** [comment]
I'd like to bring up a related issue to this. You can use "Clear all" in that right-click menu to clear current results and re-search what you were just looking for, or enable "Purge for every search" and re-search for your results again. But if I want to retain previous search results (often to compare against a newer search), there's no good way to open only the files in a particular search result. Can we get an "Open Selected File(s)" like we have for "Copy Selected Line(s)" so we can open the files associated with one search result and not ALL shown search results?

I had lots and lots of searches left over in my history and NP++ spent ~5 minutes trying to find and open them all, sometimes with the UI going unresponsive as it added new tabs to my multi-line tab view when it opened new docs, and often blocking the operation with dialog boxes telling me it couldn't find old files that no longer existed, or that e.g. "new7 doesn't exist. Would you like to create it?". There was no way to cancel all these queued up open operations, so I had to live with the pain of constantly dismissing these dialogs and waiting on the UI to catch up until it was finally done trying to open several hundred docs, and then use Close Multiple Tabs to the Right to close all those documents I never wanted to open up in the first place.

**Comment by alankilborn** [comment]
@vijaysitaram 

Looking at the history of this issue, I was proposing a workaround to @mididoc , not a solution.  It was mididoc's choice to close the issue.  It should have been left open.

It should be possible to either open "all files in all previous searches" (current behavior) OR "files with hits in a specific previous single search" (proposed new behavior).  The new behavior would be initiated by right-clicking on the "Search..." output line whose files should be opened.  

Perhaps the existing behavior gets its menu text changed from _Open all_ to "Open all hit files in all results" and a new menu item is introduced for the new behavior, possibly "Open all hit files in single result".

---

>... dialog boxes telling me it couldn't find old files that no longer existed, or that e.g. "new7 doesn't exist. Would you like to create it?"

This is bad (buggy) behavior but it deserves its own separate issue.  I will create one.
